### ISSUE TITLE
Further updates to the workflow that trains neural network, to avoid overfitting

### ISSUE THREAD

**Issue by RemiLehe** [BLAST-AI-ML/synapse#66]
- [x] Automate the number of epochs (e.g. using early stopping)
- [x] Use random sample of the data as validation set
